package-res/resources/web/pentaho/type/complex.js

```diff
@@ -616,14 +616,22 @@ define([
         var spec = {};
 
         if(requireType) {
```

Comment:
When requireType is true, the preferPropertyArray cannot be honored. This is by design.

`spec._` should always be filled in with the result of toReference.